pkg/controller.v1alpha3/suggestion/suggestionclient/suggestionclient.go

```diff
@@ -16,3 +26,172 @@ type SuggestionClient interface {
 		e *experimentsv1alpha3.Experiment,
 		ts []trialsv1alpha3.Trial) error
 }
+
+type General struct {
+}
+
+func New() SuggestionClient {
+	return &General{}
+}
+
+func (g *General) SyncAssignments(
+	instance *suggestionsv1alpha3.Suggestion,
+	e *experimentsv1alpha3.Experiment,
+	ts []trialsv1alpha3.Trial) error {
+	logger := log.WithValues("Suggestion", types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()})
+	requestNum := int(instance.Spec.Requests) - len(instance.Status.Suggestions)
+	if requestNum <= 0 {
+		return nil
+	}
+
+	endpoint := fmt.Sprintf("%s:%d", instance.Name, consts.DefaultSuggestionPort)
+	conn, err := grpc.Dial(endpoint, grpc.WithInsecure())
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	client := suggestionapi.NewSuggestionClient(conn)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	request := &suggestionapi.GetSuggestionsRequest{
+		Experiment:    g.ConvertExperiment(e),
+		Trials:        g.ConvertTrials(ts),
+		RequestNumber: int32(requestNum),
+	}
+	response, err := client.GetSuggestions(ctx, request)
+	logger.V(0).Info("Getting suggestions", "endpoint", endpoint, "response", response, "request", request)
```

Comment:
I have no idea what happened here if err != nil since I don't know what response looks like in such case
for safe, I suggest we can move line 64 in line 68